app/src/main/java/com/example/music/Presenter/MainPresenter.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.music.Presenter
+
+import android.content.Context
+import com.example.music.UI.MainView
+import com.example.music.Utils.DataMusicList
+import com.example.music.Utils.Entity.Music
+import com.example.music.Utils.LoadDataListener
+
+class MainPresenter : LoadDataListener {
+    private var mainView: MainView? = null
+    private var dataMusicList: DataMusicList? = null
+    private var context: Context? = null
+
+    constructor(mainView: MainView?, context: Context?) {
+        this.mainView = mainView
+        this.context = context
+        dataMusicList = DataMusicList(context, this)
+    }
+
+    fun loadData() {
+        dataMusicList!!.getData()
```

Comment:
không dùng !!, vì như này nó sẽ bỏ qua check null => Vẫn throws NullPointerException như binfh thường 
```suggestion
        dataMusicList?.getData()
```